hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ReplicationManager.java

```diff
@@ -1308,15 +1323,19 @@ private void handleOverReplicatedContainer(final ContainerInfo container,
           r.getDatanodeDetails().getPersistedOpState() !=
               HddsProtos.NodeOperationalState.IN_SERVICE);
 
+      // If there are unhealthy replicas, then we should remove them even if it
+      // makes the container violate the placement policy, as excess unhealthy
+      // containers are not really useful. It will be corrected later as a
+      // mis-replicated container will be seen as under-replicated.
+      // Update - delete only those unhealthy replicas which have bcsId <
+      // container bcsId
+
       final List<ContainerReplica> unhealthyReplicas = eligibleReplicas
           .stream()
           .filter(r -> !compareState(container.getState(), r.getState()))
+          .filter(r -> r.getSequenceId() > container.getSequenceId())
```

Comment:
Maybe a typo, should be r.getSequenceId() < container.getSequenceId() as the comments stated?

Reply:
Yes. Thanks @guihecheng for catching this.